### PR TITLE
Token-entry UI

### LINKED ISSUE #93: UI for entering token
## Description

Create UI screen to enable invited users to enter their token to join a project.

## Criteria

- [x] to join a project, the users should first login with their Google account
- [x] after logging in, invited users should be able to enter their token in a text field
- [x] once token is is entered in text field, the token should be validated by the backend
- [x] if the token is valid, the user should be navigated to the overview screen
- [x] once the token is validated it is linked to the user in the database and is marked as used

### PR DESCRIPTION
**Pull request by noa-drn**
# feat: Implement Token Entry Screen with validation flow

  Resolves #93
  Related to #88 (implemented in PR #133)

  ---

  ## Description

  This PR implements the token entry UI layer for the invitation system, completing the user-facing component of the token-based project access control initiated in #88.

  **User Flow:**
  1. User authenticates via Google Sign-In
  2. User enters invitation token in new token entry screen
  3. System validates token against Firestore (using InvitationRepository from #88)
  4. On success: token marked as used, user navigates to overview screen
  5. On failure: error message displayed, user can retry

  **Architecture:**
  Implements a token validation screen that gates application access behind invitation token validation. The token entry screen acts as an intermediate step between Google authentication and full application access.

  ---

  ## Changes Made

  ### New Files

  **UI Components:**
  * `app/src/main/java/ch/eureka/eurekapp/ui/authentication/TokenEntryScreen.kt` (179 lines)
    - Material 3 Compose UI with token input field
    - Real-time validation states (loading, error, success)
    - Toast-based user feedback
    - Test tags for E2E testing

  **Business Logic:**
  * `app/src/main/java/ch/eureka/eurekapp/ui/authentication/TokenEntryViewModel.kt` (142 lines)
    - StateFlow-based state management (lines 23-29: TokenEntryUIState data class)
    - Token validation workflow
    - Firestore transaction integration via InvitationRepository
    - Firebase Auth integration for user identification

  **Test Suites (with the help of Claude code):**
  * `app/src/test/java/ch/eureka/eurekapp/ui/authentication/TokenEntryViewModelTest.kt` (567 lines)
    - 34 unit tests using MockK
    - Covers initialization, validation, error handling, edge cases

  * `app/src/androidTest/java/ch/eureka/eurekapp/ui/authentication/TokenEntryScreenTest.kt`
    - UI component integration tests

  * `app/src/androidTest/java/ch/eureka/eurekapp/ui/authentication/TokenEntryViewModelFirestoreTest.kt`
    - Firestore integration tests with emulator

  * `app/src/androidTest/java/ch/eureka/eurekapp/ui/authentication/TokenEntryViewModelIntegrationTest.kt`
    - End-to-end validation workflows

  ---

Screenshot 2025-10-16 at 23 07 26

### CODE REVIEW

File: app/src/main/java/ch/eureka/eurekapp/MainActivity.kt [hunk 1]
```diff
@@ -47,6 +47,13 @@ class MainActivity : ComponentActivity() {
   }
 }
 
+/** Navigation state for the authentication flow. */
+enum class AuthState {
```
Comment (ArluAe) [inline app/src/main/java/ch/eureka/eurekapp/MainActivity.kt]: Would recommend putting 'AuthState' in another file authstate.kt

**Review by ArluAe — CHANGES_REQUESTED** [review]
Great work on this PR, the new token entry flow is cleanly designed. The UI is well structured with proper state handling and relevant testing suite.

Only asked for a change in an exception throw and to seperate an enum in another file for readability.

**Review by ArluAe — APPROVED** [review]
LGTM - All changes applied
